src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -186,11 +189,70 @@ public DynamoView(DynamoViewModel dynamoViewModel)
                  }
              };
 
+            this.HideOrShowRightSideBar();
+
             this.dynamoViewModel.RequestPaste += OnRequestPaste;
             this.dynamoViewModel.RequestReturnFocusToView += OnRequestReturnFocusToView;
             FocusableGrid.InputBindings.Clear();
         }
 
+        // This method adds a tab item to the right side bar and 
+        // sets the extension window as the tab content.
+        internal TabItem AddTabItem(Window window)
+        {
+            int count = _tabItems.Count;
+
+            tabDynamic.DataContext = null;
+
+            // creates a new tab item
+            TabItem tab = new TabItem();
+            tab.Header = window.Title;
+            tab.HeaderTemplate = tabDynamic.FindResource("TabHeader") as DataTemplate;
+
+            // setting the extension window to the current tab content
+            tab.Content = window.Content;
+
+            //Insert the tab at the end
+            _tabItems.Insert(count, tab);
+
+            tabDynamic.DataContext = _tabItems;
+            tabDynamic.SelectedItem = tab;
+
+            this.HideOrShowRightSideBar();
+
+            return tab;
+        }
+
+        // This method triggers the close operation on the selected tab. 
+        private void CloseTab(object sender, RoutedEventArgs e)
+        {
+            string tabName = (sender as Button).CommandParameter.ToString();
+            
+            TabItem tab = tabDynamic.SelectedItem as TabItem;
+
+            if (tab != null)
+            {
+                // get the selected tab
+                TabItem selectedTab = tabDynamic.SelectedItem as TabItem;
+
+                // clear tab control binding and bind to the new tab-list. 
+                tabDynamic.DataContext = null;
+                _tabItems.Remove(tab);
+                tabDynamic.DataContext = _tabItems;
+
+                // Highlight previously selected tab. if that is removed then Highlight the first tab
+                if (selectedTab == null || selectedTab.Equals(tab))
+                {
+                    if (_tabItems.Count > 0) {
+                        selectedTab = _tabItems[0];
+                    }
+                }
+                tabDynamic.SelectedItem = selectedTab;
+            }
+
+            this.HideOrShowRightSideBar();
```

Comment:
hmm - I would except this to be triggered when the items list changes, not manually here.

Reply:
Modified this and added an event when the list of tab items is modified.

Reply:
👍